lib/improver/blending/weights.py

```diff
@@ -238,12 +242,575 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+        """Create a cube to contain the output of the interpolation.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the coordinate information that will be used
+                for setting up the new_weights_cube.
+            weights (np.ndarray):
+                Weights calculated following interpolation.
+
+        Returns:
+            new_weights_cube (iris.cube.Cube):
+                Cube containing the output from the interpolation. This
+                has been renamed using the self.weights_coord_name but
+                otherwise matches the input cube.
+        """
+        cubelist = iris.cube.CubeList([])
+        for cube_slice, weight in (
+                zip(cube.slices_over(self.weighting_coord_name), weights)):
+            cube_slice.data = np.full_like(cube_slice.data, weight)
+            cubelist.append(cube_slice)
+        new_weights_cube = (
+            check_cube_coordinates(cube, cubelist.merge_cube()))
+        new_weights_cube.rename(self.weights_coord_name)
+        return new_weights_cube
+
+    def _interpolate_to_create_weights(self, cube):
+        """Method to wrap the calls to other methods to support calculation
+        of the weights by interpolation.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the coordinate information that will be used
+                for setting up the interpolation and create the new weights
+                cube.
+
+        Returns:
+            new_weights_cube (iris.cube.Cube):
+                Cube containing the output from the interpolation. This
+                has been renamed using the self.weights_coord_name but
+                otherwise matches the input cube.
+        """
+        source_points, target_points, source_weights, fill_value = (
+            self._arrange_interpolation_inputs(cube))
+        weights = WeightsUtilities.interpolate_to_find_weights(
+            source_points, target_points, source_weights,
+            fill_value=fill_value)
+        new_weights_cube = self._create_new_weights_cube(cube, weights)
+        return new_weights_cube
+
+    def process(self, cube):
+        """Calculation of linear weights based on an input configuration
+        dictionary. Weights are calculated for individually for each
+        point in self.config_coord_name. Weights are normalised across
+        the cube dimension specified by self.config_coord_name.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the coordinate information that will be used
+                for setting up the interpolation and create the new weights
+                cube.
+
+        Returns:
+            new_weights_cube (iris.cube.Cube):
+                Cube containing the output from the interpolation. This
+                has been renamed using the self.weights_coord_name but
+                otherwise matches the input cube.
```

Comment:
This is ambiguous - please either say just `Cube containing the output from the interpolation` or be specific that the cube metadata are based on the input cube.  Which may not be fully appropriate anyway (see comment above).